src/lambda.ts

```diff
@@ -0,0 +1,112 @@
+import {ListObjectsOutput} from 'aws-sdk/clients/s3';
+import {SendEmailRequest, SendEmailResponse} from 'aws-sdk/clients/ses';
+import {Config} from './config';
+import S3 = require("aws-sdk/clients/s3");
+
+let http = require('http');
+let AWS = require('aws-sdk');
+let s3 = new AWS.S3();
+let ses = new AWS.SES({region: 'eu-west-1'});
+let config = new Config();
+
+type Url = string
+type PublicationInfo = {
+    articleCount: number
+    imageCount: number
+}
+
+/**
+ * The AWS Lambda handler function.
+ * Runs a series of checks against today's Kindle publication, then sends an email
+ */
+export async function handler(): Promise<SendEmailResponse> {
+    return getRedirect(config.ManifestURL)
+        .then(testRedirect)
+        .then(() => getS3Objects(config.KindleBucket, `${config.Stage}/${config.Today}`))
+        .then(validatePublicationInfo)
+        .then(sendSuccessEmail)
+        .catch(sendFailureEmail)
+}
+
+//Returns the redirect url, and also checks that it contains today's date
+let getRedirect = (url: Url): Promise<Url> => {
+    return new Promise((resolve, reject) => {
+        http.get(url, response => {
```

Comment:
Could you use a `head` request instead?

Reply:
👍 good point

Reply:
wow, making other requests is way less convenient https://nodejs.org/api/http.html#http_http_request_options_callback

Reply:
😵